### PR TITLE
Array support 6 UUID

### PR DESCRIPTION
**Pull request by voyager-dw**
add uuid/geopoint type support in dimension and scratchspace/binder

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,8 +37,8 @@ install:
   - ./build/cmake/bin/cmake .
   - .travis/install_golang_tools.sh
 script:
-  - make lint -j
-  - make travis
+  - travis_wait 30 make lint -j
```
Comment (jshencode) [inline .travis.yml]: curious what is wait for?

File: query/aql_compiler.go [hunk 2]
```diff
@@ -766,14 +766,22 @@ func (qc *AQLQueryContext) Rewrite(expression expr.Expr) expr.Expr {
 				e.RHS = expr.Cast(e.RHS, highestType)
 			}
 
-			if rhs != nil && lhs.DataType == memCom.GeoPoint {
+			if rhs != nil && e.LHS.Type() == expr.GeoPoint {
```
Comment (jshencode) [inline query/aql_compiler.go]: why can't we just use lhs.DataType

File: query/aql_compiler.go [hunk 3]
```diff
@@ -766,14 +766,22 @@ func (qc *AQLQueryContext) Rewrite(expression expr.Expr) expr.Expr {
...
 				if val, err := memCom.GeoPointFromString(rhs.Val); err != nil {
 					qc.Error = err
 				} else {
 					e.RHS = &expr.GeopointLiteral{
 						Val: val,
 					}
 				}
+			} else if rhs != nil && e.LHS.Type() == expr.UUID {
```
Comment (jshencode) [inline query/aql_compiler.go]: same here

File: query/aql_compiler.go [hunk 4]
```diff
@@ -1108,6 +1116,20 @@ func (qc *AQLQueryContext) Rewrite(expression expr.Expr) expr.Expr {
...
+				case memCom.UUID:
+					strLiteral, ok := secondArg.(*expr.StringLiteral)
+					if !ok {
+						qc.Error = utils.StackError(nil, "array function %s needs uuid string literal", e.Name)
+						break
+					}
+					if val, err := memCom.UUIDFromString(strLiteral.Val); err != nil {
+						qc.Error = err
+						break
+					} else {
```
Comment (jshencode) [inline query/aql_compiler.go]: nit: else seems unnecessary

File: query/iterator.hpp [hunk 5]
```diff
@@ -565,7 +566,14 @@ inline typename ConstantIterator<UUIDT>::type make_constant_iterator(
   return thrust::make_constant_iterator(
       thrust::make_tuple<UUIDT, bool>(defaultValue, defaultNull));
 }
-
+/*
+template <>
+inline typename ConstantIterator<GeoPointT>::type make_constant_iterator(
```
Comment (jshencode) [inline query/iterator.hpp]: remove if not needed

File: .travis.yml (hunk 1 above)
Comment (voyager-dw) [reply .travis.yml]: updated with new method, this intends to mitigate timeout issue

File: query/aql_compiler.go (hunk 2 above)
Comment (voyager-dw) [reply query/aql_compiler.go]: lhs is nil sometimes

File: query/aql_compiler.go (hunk 3 above)
Comment (voyager-dw) [reply query/aql_compiler.go]: lhs is not always var type, so it could be nil

File: .travis.yml (hunk 1 above)
Comment (jshencode) [reply .travis.yml]: might need to check how much slow down in the c code compilation

Comment (voyager-dw) [reply .travis.yml]: yeah, had a ticket created to improve it, most time is on the scratch_transform and dimension_transform, as we added output types from 3 to 5.

**Pull request merged**